invenio_app_ils/permissions.py

```diff
@@ -134,25 +134,30 @@ def patron_owner_permission(record):
 def loan_checkout_permission(*args, **kwargs):
     """Return permission to allow admins and librarians to checkout and patrons to self-checkout if enabled."""
     if not has_request_context():
-        # If from CLI, don't allow self-checkout
+        # CLI or Celery task
         return backoffice_permission()
+
     if current_user.is_anonymous:
         abort(401)
 
     is_admin_or_librarian = backoffice_permission().allows(g.identity)
     if is_admin_or_librarian:
         return backoffice_permission()
+
+    # ensure that only the loan's patron can do operations on this loan
     if len(args):
         loan = args[0]
     else:
-        loan = kwargs.get("record", {})
-    is_patron_current_user = current_user.id == int(loan.get("patron_pid"))
+        loan = kwargs["record"]
+    is_patron_current_user = current_user.id == int(loan["patron_pid"])
```

Comment:
We should really avoid using `.get` when the dict field is required. We need to fail early in these cases.